search-resources-api/src/main/java/no/unit/nva/search/SearchTicketsHandler.java

```diff
@@ -39,153 +39,166 @@
-  }
-
-  private String getIndexName(RequestInfo requestInfo) {
-    String requestPath = RequestUtil.getRequestPath(requestInfo);
-    return UnixPath.of(requestPath).getLastPathElement();
-  }
+    public static final String VIEWING_SCOPE_QUERY_PARAMETER = "viewingScope";
+    public static final String CRISTIN_ORG_LEVEL_DELIMITER = "\\.";
+    public static final int HIGHEST_LEVEL_ORGANIZATION = 0;
+    public static final String ACCESS_RIGHTS_TO_VIEW_TICKETS = "APPROVE_DOI_REQUEST";
+    public static final String ROLE_CURATOR = "curator";
+    public static final String PARAM_ROLE = "role";
+    public static final String ROLE_CREATOR = "creator";
+    private static final Logger logger = LoggerFactory.getLogger(SearchTicketsHandler.class);
+    private final SearchClient searchClient;
+    private final IdentityClient identityClient;
+
+    @JacocoGenerated
+    public SearchTicketsHandler() {
+        this(new Environment(), defaultSearchClient(), defaultIdentityClient());
+    }
+
+    public SearchTicketsHandler(Environment environment, SearchClient searchClient, IdentityClient identityClient) {
+        super(Void.class, environment, objectMapperWithEmpty);
+        this.searchClient = searchClient;
+        this.identityClient = identityClient;
+    }
+
+    @Override
+    protected SearchResponseDto processInput(Void input, RequestInfo requestInfo, Context context)
+        throws ApiGatewayException {
+        var indexName = getIndexName(requestInfo);
+        logger.info("Index name: {}", indexName);
+        return handleSearch(requestInfo, indexName);
+    }
+
+    @Override
+    protected Integer getSuccessStatusCode(Void input, SearchResponseDto output) {
+        return HTTP_OK;
+    }
+
+    private static BoolQueryBuilder notInFilter(RequestInfo requestInfo, String owner, String mustField,
+                                                String notInField) {
+        return new BoolQueryBuilder()
+                   .must(QueryBuilders.queryStringQuery(RequestUtil.getSearchTerm(requestInfo)))
+                   .must(QueryBuilders.matchQuery(mustField, owner).operator(Operator.AND))
+                   .mustNot(QueryBuilders.matchQuery(notInField, owner).operator(Operator.AND));
+    }
+
+    @JacocoGenerated
+    private static IdentityClient defaultIdentityClient() {
+        return new IdentityClientImpl();
+    }
+
+    private static boolean userHasAccessRights(RequestInfo requestInfo) {
+        return requestInfo.userIsAuthorized(ACCESS_RIGHTS_TO_VIEW_TICKETS);
+    }
+
+    private SearchResponseDto handleSearch(RequestInfo requestInfo, String indexName)
+        throws ApiGatewayException {
+        var role = requestInfo.getQueryParameterOpt(PARAM_ROLE).orElse(ROLE_CURATOR);
+        if (ROLE_CURATOR.equals(role)) {
+            assertCuratorHasAppropriateAccessRights(requestInfo);
+            return handleCuratorSearch(requestInfo, indexName);
+        }
+        if (ROLE_CREATOR.equals(role)) {
```

Comment:
ROLE_CREATOR.equals seems redundant?

if you are not a creator, you don't have any tickets and then you will receive an empty list.